scraper/outputs/elasticsearch_output.py

```diff
@@ -0,0 +1,89 @@
+from elasticsearch import AsyncElasticsearch, NotFoundError
+from elasticsearch.helpers import async_bulk
+from datetime import datetime
+from typing import List
+from loguru import logger
+
+from scraper.models import MetadataDocument, ScrapedDocument
+from scraper.outputs.abstract_output import AbstractOutput
+from scraper.config import SourceConfig, settings
+
+
+class ElasticsearchOutput(AbstractOutput):
+    def __init__(self):
+        self.es = None
+        self.metadata_prefix = "scrape_metadata_"
+
+    async def initialize(self):
+        try:
+            self.es = AsyncElasticsearch(
+                cloud_id=settings.CLOUD_ID, api_key=settings.API_KEY
+            )
+        except Exception as e:
+            logger.error(f"Failed to initialize Elasticsearch: {e}")
+            raise
+
+    async def close(self):
+        if self.es:
+            await self.es.close()
+
+    async def index_documents(self, documents: List[ScrapedDocument], index_name: str):
+        try:
+            actions = [
+                {
+                    "_op_type": "index",
+                    "_index": index_name,
+                    "_id": doc.id,
+                    "_source": doc.model_dump(exclude_none=True),
+                }
+                for doc in documents
+            ]
+
+            success, failed = await async_bulk(self.es, actions, raise_on_error=False)
```

Comment:
@kouloumos Here, we would need to add one more attribute so that we do not run into Timeout Error while indexing the data:
`success, failed = await async_bulk(self.es, actions, raise_on_error=False, request_timeout=120)`

Reply:
good catch! Do we still need that if I change the code to process documents in batches? Something like 500 documents per batch.

Reply:
Can't confirm that for sure, can you please test it while writing the script?

Reply:
It was needed after all, I added it.